Sources/SwiftSentry/User.swift

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: BSD-3-Clause
+import Foundation
+
+/// The representation of a user which should be attached to subsequent events.
+public struct User {
+    var userId: String?
+    var email: String?
+    var username: String?
+
+    public init(
+        userId: String? = nil,
+        email: String? = nil,
+        username: String? = nil
+    ) {
+      self.userId = userId
+      self.email = email
+      self.username = username
+    }
+}
+
+extension User: SentryValueSerializable {
+    func serialized() -> sentry_value_t {
```

Comment:
Mind giving this explicit access control?

Reply:
Marked this and the protocol explicitly as `internal` since marking them as `private` prevents the ability to get access to them with a testable import.